python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -768,6 +769,16 @@ def _validate_input(self):
                 raise NotImplementedError(
                     "Slice only supports literal start and stop values"
                 )
+        elif self.name == pl_expr.StringFunction.Strptime:
+            format, strict, exact, cache = self.options
+            if cache:
+                raise NotImplementedError("Strptime cache is a CPU feature")
+            if format is None:
+                raise NotImplementedError("Strptime format is required")
+            if not exact:
+                raise NotImplementedError("Strptime does not support exact=False")
+            if strict:
+                raise NotImplementedError("Strptime does not support strict=True")
```

Comment:
Why do we not support strict=True? Can we not attempt to parse (via `is_timestamp`) and raise if there are any nulls?

Reply:
I was going to ask about this pattern more broadly as it came up in conversation last week. What's the pattern we want to establish for handling runtime errors on the GPU side? Usually if we fail, it happens at query planning time, allowing for fallback, but here we might have a runtime error. Would we want to fail and raise the same error as polars? Or fall back and have polars run the query and presumably error in the same spot, directly from the polars side?

Reply:
If we can error in the same way (does not need to raise the same error, I think) as polars would, then we should not fallback at query planning time.

IOW: it is legitimate for us to complain "can't do that" at runtime if polars-CPU would also complain similarly.

Reply:
I think we need a testing utility which asserts both the cuDF and polars side raise at runtime, or to modify `assert_gpu_result_equal` to handle error cases since it collects the CPU side first and won't make it to the GPU side if that errors, working on this.

Reply:
I added a utility for this. The one thing I'm struggling with is that I'd like to enforce that both sides raise the same kind of exception, in this case an `InvalidOperationError` from polars. But the way things work in cudf_polars currently the error ends up wrapped in a `ComputeError` regardless of the exception raised on the GPU side. Do we want to enforce the exceptions are the same at all?

Reply:
The above is also why a particular test currently fails CI.

Reply:
> Do we want to enforce the exceptions are the same at all?

No